client/src/main/kotlin/io/spine/examples/chatspn/desktop/Navigation.kt

```diff
@@ -26,48 +26,692 @@
+    val selectedChat = model.selectedChat()
+    LazyColumn(
+        Modifier.fillMaxSize()
+    ) {
+        chats.forEachIndexed { index, chat ->
+            item(key = index) {
+                ChatPreviewPanel(
+                    chat.name,
+                    chat.lastMessage,
+                    chat.id.equals(selectedChat.value)
+                ) {
+                    model.selectChat(chat.id)
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Represents the chat preview in the chats list.
+ */
+@Composable
+private fun ChatPreviewPanel(
+    chatName: String,
+    lastMessage: MessageData?,
+    isSelected: Boolean,
+    select: () -> Unit
+) {
+    Box(
+        Modifier
+            .fillMaxWidth()
+            .clickable { select() }
+            .background(
+                color = if (isSelected)
+                    MaterialTheme.colorScheme.inverseSurface
+                else
+                    MaterialTheme.colorScheme.background
+            ),
+        Alignment.CenterStart,
+    ) {
+        ChatPreviewContent(chatName, lastMessage)
+    }
+}
+
+/**
+ * Represents the chat preview content in the chat preview panel.
+ */
+@Composable
+private fun ChatPreviewContent(chatName: String, lastMessage: MessageData?) {
+    Row(
+        Modifier.padding(8.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Avatar(46f, chatName)
+        Spacer(Modifier.size(12.dp))
+        Column(horizontalAlignment = Alignment.Start) {
+            Row(
+                Modifier.fillMaxWidth(),
+                Arrangement.SpaceBetween
+            ) {
+                Text(
+                    text = chatName,
+                    color = MaterialTheme.colorScheme.onBackground,
+                    style = MaterialTheme.typography.headlineMedium,
+                )
+                Text(
+                    text = if (lastMessage == null) "" else lastMessage.whenPosted.toStringTime(),
+                    color = MaterialTheme.colorScheme.onSecondary,
+                    style = MaterialTheme.typography.bodySmall,
+                )
+            }
+            Spacer(Modifier.size(9.dp))
+            Text(
+                text = if (lastMessage == null) ""
+                else lastMessage.content.replace("\\s".toRegex(), " "),
+                color = MaterialTheme.colorScheme.onSecondary,
+                style = MaterialTheme.typography.bodyMedium,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis
+            )
+        }
+    }
+}
+
+/**
+ * Represents content when no one chat is selected.
  */
-public class Navigation(private val client: DesktopClient) {
-    private val currentPage: MutableState<Page> = mutableStateOf(Page.REGISTRATION)
+@Composable
+private fun ChatNotChosenBox() {
+    Box(
+        Modifier.fillMaxSize(),
+        Alignment.Center
+    ) {
+        Text(
+            text = "Choose the chat",
```

Comment:
"a chat".

That's another point we should be thinking about.

Let's have a task to extract all text messages into resources. The code should be ready for internationalisation (or localisation, whichever you prefer) at any time.